### PR TITLE
Added body close on retry for downloader round trip

### PR DESCRIPTION
**Pull request by mh0lt**
Add missing body close method when webseed roundtrip is retried

### CODE REVIEW

**Review by Fm8914 — APPROVED** [review]

**Review by Fm8914 — COMMENTED** [review]
Uvojpkln0l

File: erigon-lib/downloader/downloader.go [hunk 1]
```diff
@@ -183,6 +183,11 @@ func (r *requestHandler) RoundTrip(req *http.Request) (resp *http.Response, err
 		case http.StatusInternalServerError, http.StatusBadGateway:
 			r.downloader.stats.WebseedServerFails.Add(1)
 
+			if resp.Body != nil {
```
Comment (AskAlexSharov) [inline erigon-lib/downloader/downloader.go:186]: maybe also add: 
```
defer func() {
   if err == nil && resp != nil && resp.Body != nil {
        resp.Body.Close()
   }
}
```

**Review by AskAlexSharov — COMMENTED** [review]
```
if err == nil && delay < maxDelay {
					delay = delay + (time.Duration(rand.Intn(200-75)+75)*delay)/100
				}
```
did you mean `!= nil`?

**Review by AskAlexSharov — COMMENTED** [review]
FYI: you also can take a look for inspiration https://github.com/hashicorp/go-retryablehttp/blob/main/client.go#L535

**Comment by AskAlexSharov** [comment]
Plz add comment about: why this status codes are choosen, why no other retryable codes:  https://stackoverflow.com/a/74627395

File: erigon-lib/downloader/downloader.go (hunk 1 above)
Comment (mh0lt) [reply erigon-lib/downloader/downloader.go:186]: This is an intermediate function I'm not sure it should be catching panics - I think they should propagate up to the caller ?

Comment (AskAlexSharov) [reply erigon-lib/downloader/downloader.go:233]: If need propagate - can propagate by `panic(rec)`.  i'm only about closing body.

**Review by AskAlexSharov — APPROVED** [review]

**Pull request merged**